File: app/model/message/message.ts
export enum MessageType{
  TEXT = "TEXT",
  FILE = "FILE"
} 

export interface Message {
  id: string;
  content?: string;
  fileUrl?: string;
  fileName?: string;
  senderId: string;  
  sender: string;
  createdAt: string;
  type: MessageType; // <--- add this
}

File: backend/models/message.go
package models

import (
	"time"
)

type MessageType string

const (
	TEXTMSG MessageType = "TEXT"
	FILE    MessageType = "FILE"
)

type Message struct {
	ID        string      `json:"id" gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	Content   string      `json:"content,omitempty"`  // Text message
	FileURL   string      `json:"fileUrl,omitempty"`  // Cloudinary URL
	FileName  string      `json:"fileName,omitempty"` // Original filename
	Type      MessageType `json:"type"`               // TEXT | FILE
	SenderID  string      `json:"senderId"`           
	ChannelID string      `json:"channelId"`          
	CreatedAt time.Time   `json:"createdAt"`
	UpdatedAt time.Time   `json:"updatedAt"`
}
